### ISSUE TITLE
单机多卡训练 设置不同的batchsize时会导致训练开始时卡住，没有任何错误信息被记录或者打印

### ISSUE THREAD

**Issue by BossCrab-jyj** [PaddlePaddle/Paddle#57220]
### 复现环境 Environment

platform: Linux-5.15.0-82-generic-x86_64-with-glibc2.29
Python: 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
Paddle compiled with cuda: True
NVCC: Not Available
cudnn: 8.9
GPUs used: 6
CUDA_VISIBLE_DEVICES: None
GPU: ['GPU 0: NVIDIA GeForce', 'GPU 1: NVIDIA GeForce', 'GPU 2: NVIDIA GeForce', 'GPU 3: NVIDIA GeForce', 'GPU 4: NVIDIA GeForce', 'GPU 5: NVIDIA GeForce']，6张4090，显存约150gb
cpu：64核
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
PaddleSeg: 2.8.0
PaddlePaddle: 2.5.1
OpenCV: 4.5.5

训练代码：python -m paddle.distributed.launch tools/train.py        --config configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml        --do_eval        --use_vdl        --save_interval 1        --save_dir output --num_workers 12  --log_iters 1 --batch_size 12  可以正常训练。
训练代码：python -m paddle.distributed.launch tools/train.py        --config configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml        --do_eval        --use_vdl        --save_interval 1        --save_dir output --num_workers 60 --log_iters 1 --batch_size 60或者--num_workers 12  --num_workers 60 或者--num_workers 12  --batch_size 24卡住。

### 以下是运行信息：
------------------------------------------------
LAUNCH INFO 2023-09-12 11:08:21,790 -----------  Configuration  ----------------------
LAUNCH INFO 2023-09-12 11:08:21,790 auto_parallel_config: None
LAUNCH INFO 2023-09-12 11:08:21,790 devices: None
LAUNCH INFO 2023-09-12 11:08:21,790 elastic_level: -1
LAUNCH INFO 2023-09-12 11:08:21,790 elastic_timeout: 30
LAUNCH INFO 2023-09-12 11:08:21,790 gloo_port: 6767
LAUNCH INFO 2023-09-12 11:08:21,790 host: None
LAUNCH INFO 2023-09-12 11:08:21,790 ips: None
LAUNCH INFO 2023-09-12 11:08:21,790 job_id: default
LAUNCH INFO 2023-09-12 11:08:21,790 legacy: False
LAUNCH INFO 2023-09-12 11:08:21,790 log_dir: log
LAUNCH INFO 2023-09-12 11:08:21,790 log_level: INFO
LAUNCH INFO 2023-09-12 11:08:21,790 log_overwrite: False
LAUNCH INFO 2023-09-12 11:08:21,790 master: None
LAUNCH INFO 2023-09-12 11:08:21,790 max_restart: 3
LAUNCH INFO 2023-09-12 11:08:21,790 nnodes: 1
LAUNCH INFO 2023-09-12 11:08:21,790 nproc_per_node: None
LAUNCH INFO 2023-09-12 11:08:21,791 rank: -1
LAUNCH INFO 2023-09-12 11:08:21,791 run_mode: collective
LAUNCH INFO 2023-09-12 11:08:21,791 server_num: None
LAUNCH INFO 2023-09-12 11:08:21,791 servers:
LAUNCH INFO 2023-09-12 11:08:21,791 start_port: 6070
LAUNCH INFO 2023-09-12 11:08:21,791 trainer_num: None
LAUNCH INFO 2023-09-12 11:08:21,791 trainers:
LAUNCH INFO 2023-09-12 11:08:21,791 training_script: tools/train.py
LAUNCH INFO 2023-09-12 11:08:21,791 training_script_args: ['--config', 'configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml', '--do_eval', '--use_vdl', '--save_interval', '1', '--save_dir', 'output', '--num_workers', '60', '--log_iters', '1', '--batch_size', '60']
LAUNCH INFO 2023-09-12 11:08:21,791 with_gloo: 1
LAUNCH INFO 2023-09-12 11:08:21,791 --------------------------------------------------
LAUNCH INFO 2023-09-12 11:08:21,792 Job: default, mode collective, replicas 1[1:1], elastic False
LAUNCH INFO 2023-09-12 11:08:21,793 Run Pod: wbgglu, replicas 6, status ready
LAUNCH INFO 2023-09-12 11:08:21,883 Watching Pod: wbgglu, replicas 6, status running
======================= Modified FLAGS detected =======================
FLAGS(name='FLAGS_selected_gpus', current_value='0', default_value='')
=======================================================================
I0912 11:08:23.754560 2598655 tcp_utils.cc:181] The server starts to listen on IP_ANY:41136
I0912 11:08:23.754935 2598655 tcp_utils.cc:130] Successfully connected to 127.0.1.1:41136
W0912 11:08:25.743657 2598655 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 8.9, Driver API Version: 12.2, Runtime API Version: 11.7
W0912 11:08:25.744632 2598655 gpu_resources.cc:149] device: 0, cuDNN Version: 8.9.
2023-09-12 11:08:27 [WARNING]   Add the `num_classes` in train_dataset and val_dataset config to model config. We suggest you manually set `num_classes` in model config.
2023-09-12 11:08:28 [INFO]
------------Environment Information-------------
platform: Linux-5.15.0-82-generic-x86_64-with-glibc2.29
Python: 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
Paddle compiled with cuda: True
NVCC: Not Available
cudnn: 8.9
GPUs used: 6
CUDA_VISIBLE_DEVICES: None
GPU: ['GPU 0: NVIDIA GeForce', 'GPU 1: NVIDIA GeForce', 'GPU 2: NVIDIA GeForce', 'GPU 3: NVIDIA GeForce', 'GPU 4: NVIDIA GeForce', 'GPU 5: NVIDIA GeForce']
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
PaddleSeg: 2.8.0
PaddlePaddle: 2.5.1
OpenCV: 4.5.5
------------------------------------------------
2023-09-12 11:08:28 [INFO]
---------------Config Information---------------
batch_size: 60
iters: 1000
train_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: train
  num_classes: 2
  train_path: /home/jh/jyj/datasets/my_data/train_list.txt
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 512
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.5
    contrast_range: 0.5
    saturation_range: 0.5
    type: RandomDistort
  - type: Normalize
  type: Dataset
val_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: val
  num_classes: 2
  transforms:
  - type: Normalize
  type: Dataset
  val_path: /home/jh/jyj/datasets/my_data/val_list.txt
optimizer:
  momentum: 0.9
  type: SGD
  weight_decay: 4.0e-05
lr_scheduler:
  end_lr: 0
  learning_rate: 0.01
  power: 0.9
  type: PolynomialDecay
loss:
  coef:
  - 1
  - 1
  - 1
  types:
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
model:
  backbone:
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
    type: STDC2
  num_classes: 2
  type: PPLiteSeg
------------------------------------------------

2023-09-12 11:08:28 [INFO]      Set device: gpu
2023-09-12 11:08:28 [INFO]      Use the following config to build model
model:
  backbone:
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
    type: STDC2
  num_classes: 2
  type: PPLiteSeg
2023-09-12 11:08:28 [INFO]      Loading pretrained model from https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
2023-09-12 11:08:28 [INFO]      There are 265/265 variables loaded into STDCNet.
2023-09-12 11:08:28 [INFO]      Convert bn to sync_bn
2023-09-12 11:08:28 [INFO]      Use the following config to build train_dataset
train_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: train
  num_classes: 2
  train_path: /home/jh/jyj/datasets/my_data/train_list.txt
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 512
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.5
    contrast_range: 0.5
    saturation_range: 0.5
    type: RandomDistort
  - type: Normalize
  type: Dataset
2023-09-12 11:08:28 [INFO]      Use the following config to build val_dataset
val_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: val
  num_classes: 2
  transforms:
  - type: Normalize
  type: Dataset
  val_path: /home/jh/jyj/datasets/my_data/val_list.txt
2023-09-12 11:08:28 [INFO]      If the type is SGD and momentum in optimizer config, the type is changed to Momentum.
2023-09-12 11:08:28 [INFO]      Use the following config to build optimizer
optimizer:
  momentum: 0.9
  type: Momentum
  weight_decay: 4.0e-05
2023-09-12 11:08:28 [INFO]      Use the following config to build loss
loss:
  coef:
  - 1
  - 1
  - 1
  types:
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
[2023-09-12 11:08:28,762] [    INFO] distributed_strategy.py:160 - distributed strategy initialized
[2023-09-12 11:08:28,763] [ WARNING] fleet.py:290 - The dygraph parallel environment has been initialized.
[2023-09-12 11:08:29,915] [    INFO] topology.py:266 - Total 1 data comm group(s) create successfully!
[2023-09-12 11:08:29,929] [    INFO] topology.py:266 - Total 6 model comm group(s) create successfully!
[2023-09-12 11:08:29,942] [    INFO] topology.py:266 - Total 6 pipe comm group(s) create successfully!
[2023-09-12 11:08:29,956] [    INFO] topology.py:266 - Total 6 sharding comm group(s) create successfully!
[2023-09-12 11:08:29,969] [    INFO] topology.py:217 - HybridParallelInfo: rank_id: 0, mp_degree: 1, sharding_degree: 1, pp_degree: 1, dp_degree: 6, mp_group: [0],  sharding_group: [0], pp_group: [0], dp_group: [0, 1, 2, 3, 4, 5], check/clip group: [0]

程序一直卡在这里不动，单块GPU显存显示只占用 3492MiB / 24564MiB，GPU利用率为0，使用htop查看cpu一直在运行，连续观察3-5分钟，依旧卡住不动。

**Comment by GuoxiaWang** [comment]
你看看每台机器上的 log/workerlog.x 有没有日志报错

**Comment by BossCrab-jyj** [comment]
这是其中一个log的日志，其他的都是一样的，没有发现错误信息

> 

======================= Modified FLAGS detected =======================
FLAGS(name='FLAGS_selected_gpus', current_value='5', default_value='')
=======================================================================
I0912 19:11:54.165021 677678 tcp_utils.cc:107] Retry to connect to 127.0.1.1:38730 while the server is not yet listening.
I0912 19:11:57.165335 677678 tcp_utils.cc:130] Successfully connected to 127.0.1.1:38730
W0912 19:11:58.886976 677678 gpu_resources.cc:119] Please NOTE: device: 5, GPU Compute Capability: 8.9, Driver API Version: 12.2, Runtime API Version: 11.7
W0912 19:11:58.889166 677678 gpu_resources.cc:149] device: 5, cuDNN Version: 8.9.
[2023-09-12 19:12:00,493] [    INFO] distributed_strategy.py:160 - distributed strategy initialized
[2023-09-12 19:12:00,493] [ WARNING] fleet.py:290 - The dygraph parallel environment has been initialized.
[2023-09-12 19:12:00,830] [    INFO] topology.py:266 - Total 1 data comm group(s) create successfully!
[2023-09-12 19:12:00,832] [    INFO] topology.py:266 - Total 6 model comm group(s) create successfully!
[2023-09-12 19:12:00,833] [    INFO] topology.py:266 - Total 6 pipe comm group(s) create successfully!
[2023-09-12 19:12:00,833] [    INFO] topology.py:266 - Total 6 sharding comm group(s) create successfully!
[2023-09-12 19:12:00,834] [    INFO] topology.py:217 - HybridParallelInfo: rank_id: 5, mp_degree: 1, sharding_degree: 1, pp_degree: 1, dp_degree: 6, mp_group: [5],  sharding_group: [5], pp_group: [5], dp_group: [0, 1, 2, 3, 4, 5], check/clip group: [5]

**Comment by GuoxiaWang** [comment]
所有机器上都一样的吗？能多等一会儿吗？因为有可能是在加载数据之类的。

看你描述的，不像是 hang 了。 gpu 利用率才是0.

**Comment by BossCrab-jyj** [comment]
我是单机多卡，所有log文件都是这样的，从刚刚回答您已经过去6分钟了，log文件和命令终端依旧没有变化，
这是显卡情况
Tue Sep 12 19:25:09 2023
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.86.05              Driver Version: 535.86.05    CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4090        Off | 00000000:17:00.0 Off |                  Off |
| 43%   26C    P8              14W / 450W |    740MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 4090        Off | 00000000:31:00.0 Off |                  Off |
| 46%   28C    P8              26W / 450W |    740MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   2  NVIDIA GeForce RTX 4090        Off | 00000000:4B:00.0 Off |                  Off |
| 44%   27C    P8              22W / 450W |    740MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   3  NVIDIA GeForce RTX 4090        Off | 00000000:B1:00.0 Off |                  Off |
| 45%   29C    P8              15W / 450W |    740MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   4  NVIDIA GeForce RTX 4090        Off | 00000000:CA:00.0 Off |                  Off |
| 45%   29C    P8              18W / 450W |    740MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   5  NVIDIA GeForce RTX 4090        Off | 00000000:E3:00.0 Off |                  Off |
| 44%   26C    P8              22W / 450W |    740MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      2599      G   /usr/lib/xorg/Xorg                            4MiB |
|    0   N/A  N/A    677666      C   ...jh/.virtualenvs/ppaddle/bin/python3      720MiB |
|    1   N/A  N/A      2599      G   /usr/lib/xorg/Xorg                            4MiB |
|    1   N/A  N/A    677668      C   ...jh/.virtualenvs/ppaddle/bin/python3      720MiB |
|    2   N/A  N/A      2599      G   /usr/lib/xorg/Xorg                            4MiB |
|    2   N/A  N/A    677670      C   ...jh/.virtualenvs/ppaddle/bin/python3      720MiB |
|    3   N/A  N/A      2599      G   /usr/lib/xorg/Xorg                            4MiB |
|    3   N/A  N/A    677672      C   ...jh/.virtualenvs/ppaddle/bin/python3      720MiB |
|    4   N/A  N/A      2599      G   /usr/lib/xorg/Xorg                            4MiB |
|    4   N/A  N/A    677674      C   ...jh/.virtualenvs/ppaddle/bin/python3      720MiB |
|    5   N/A  N/A      2599      G   /usr/lib/xorg/Xorg                            4MiB |
|    5   N/A  N/A    677678      C   ...jh/.virtualenvs/ppaddle/bin/python3      720MiB |
+---------------------------------------------------------------------------------------+
可以看到paddle进程还在。

**Comment by juncaipeng** [comment]
你好，使用paddleseg的示例数据集，是否可以跑通？

**Comment by BossCrab-jyj** [comment]
示例数据集我没有跑，直接训练的自己的数据集，使用这个命令是可以正常训练的，并且测试了模型，效果还行，应该和数据没有关系。
python -m paddle.distributed.launch tools/train.py --config configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml --do_eval --use_vdl --save_interval 1 --save_dir output --num_workers 12 --log_iters 1 --batch_size 12

**Comment by juncaipeng** [comment]
嗯，跑不起来的启动脚本是怎样啊？

**Comment by BossCrab-jyj** [comment]
刚刚试了示例数据集也是不行，使用的是
python -m paddle.distributed.launch tools/train.py --config configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml --do_eval --use_vdl --save_interval 1 --save_dir output --num_workers 12 --log_iters 1 --batch_size 60
只是更改了batchsize，很奇怪改回 --batch_size 12 示例数据集也可以跑了

**Comment by juncaipeng** [comment]
num_workers建议减小点

**Comment by BossCrab-jyj** [comment]
我的cpu有64核，num_workers设置4也是不行，我试了下，batch_size 小于等于12是可以的，大于就不行
